sklearn/utils/__init__.py

```diff
@@ -47,7 +49,7 @@ class deprecated(object):
     # Adapted from http://wiki.python.org/moin/PythonDecoratorLibrary,
     # but with many changes.
 
-    def __init__(self, extra=''):
+    def __init__(self, extra='', msg=None, update_doc=True):
```

Comment:
What is is the usecase for this new `msg` parameter? Do we actually need it anywhere in the code base?